app/src/main/java/org/thoughtcrime/securesms/search/SearchRepository.java

```diff
@@ -123,23 +102,20 @@ public void query(@NonNull String query, @NonNull Callback<SearchResult> callbac
   }
 
   public void query(@NonNull String query, long threadId, @NonNull Callback<CursorList<MessageResult>> callback) {
-    if (TextUtils.isEmpty(query)) {
+    // If the sanitized search query is empty or less than 2 chars then abort the search
+    String cleanQuery = sanitizeQuery(query).trim();
+    if (cleanQuery.isEmpty() || cleanQuery.length() < 2) {
```

Comment:
a bit tautological, `length < 2` would cover both cases, but maybe the `isEmpty` is preferred?

Reply:
Adjusted so only using sanitize -> trim -> isEmpty